### PR TITLE
docs: Spell out "Install from Git repository" options properly

### PR DESCRIPTION
**Pull request by MichalBryxi**
- Create sections to make visual/logical distinction of individual parts.
- Spell out all major possible syntaxes.
- Add more explanation to individual parts of the `URL` like `semver:` and `path:`.
- Copy & paste examples directly from [tests](https://github.com/pnpm/pnpm/blob/main/resolving/git-resolver/test/index.ts) so that we're reasonably sure they are correct.
- Explain defaults.
- Explain common cases like monorepo & forks.

Fixes: #577

### CODE REVIEW

**Comment by MichalBryxi** [comment]
Happy to talk about any and all choices and alter things if necessary, but I firmly believe that this amount of veboseness greatly reduces user confusion as they don't have to guess what is possible / why / how.

File: docs/cli/add.md [hunk 1]
```diff
@@ -81,26 +81,87 @@ pnpm add <git remote url>
...
+```
+
+#### Providers shorthand
+
+You can use a protocol shorthand `[provier]:` for certain Git providers:
+
+```
+pnpm add github:zkochan/is-negative
+pnpm add bitbucket:pnpmjs/git-resolver
+pnpm add gitlab:pnpm/git-resolver
```
Comment (MichalBryxi) [inline docs/cli/add.md:152]: Is there more? I found only those in tests.

File: docs/cli/add.md [hunk 2]
```diff
@@ -81,26 +81,87 @@ pnpm add <git remote url>
...
-`pnpm add github:user/repo`
 
-You may install from Git by:
+You may install packages from Git by:
 
-* latest commit from master: `pnpm add kevva/is-positive`
-* commit: `pnpm add kevva/is-positive#97edff6f525f192a3f83cea1944765f769ae2678`
-* branch: `pnpm add kevva/is-positive#master`
-* version range: `pnpm add kevva/is-positive#semver:^2.0.0`
+```
```
Comment (MichalBryxi) [inline docs/cli/add.md:87]: Personally I prefer these triple-fenced examples, as they are way easier to target with a mouse for copy&pasting as you're constrained by the newlines. As opposed to the bullet point format above, which allows for copying the non-code prefix as well.

File: docs/cli/add.md [hunk 3]
```diff
@@ -81,26 +81,87 @@ pnpm add <git remote url>
...
+pnpm add zkochan/is-negative#2.0.1
+
+# v-prefixed git tag
+pnpm add andreineculau/npm-publish-git#v0.0.
+```
+
+### Semver
 
-You may also install just a subdirectory from a Git-hosted monorepo. For instance:
+You can specify version (range) to install using the `semver:` parameter. For example:
```
Comment (MichalBryxi) [inline docs/cli/add.md:109]: I consider the `semver:` and `path:` (below) being explicitly spelled out very important. It makes sure that the user knows it's a keyword and not just a random thing in given repo.

File: docs/cli/add.md [hunk 4]
```diff
@@ -81,26 +81,87 @@ pnpm add <git remote url>
...
 
-You may install from Git by:
+You may install packages from Git by:
 
-* latest commit from master: `pnpm add kevva/is-positive`
-* commit: `pnpm add kevva/is-positive#97edff6f525f192a3f83cea1944765f769ae2678`
-* branch: `pnpm add kevva/is-positive#master`
-* version range: `pnpm add kevva/is-positive#semver:^2.0.0`
+```
+# latest commit from default branch
```
Comment (MichalBryxi) [inline docs/cli/add.md:88]: As far as I can tell `pnpm` does not look for `master` per-se, it looks for (in GitHub terminology) [default branch](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch), hence the naming change here.

File: docs/cli/add.md (hunk 2 above)
Comment (zkochan) [reply docs/cli/add.md:87]: you can have both. code sections inside bullet points. I don't think there's a reason to move the descriptions into the code blocks.

File: docs/cli/add.md [hunk 5]
```diff
@@ -81,26 +81,87 @@ pnpm add <git remote url>
...
+
+You may also install just a subdirectory from a Git-hosted monorepo using the `path:` parameter. For instance:
+
 ```
+pnpm add RexSkz/test-git-subfolder-fetch#path:/packages/simple-react-app
+```
+
+#### Full URL
+
+If you want to be more explicit or are using alternative git hosting, you might want to spell out full git URL:
```
Comment (zkochan) [inline docs/cli/add.md:135]: Git should be capitilized
```suggestion
If you want to be more explicit or are using alternative Git hosting, you might want to spell out full Git URL:
```

File: docs/cli/add.md (hunk 2 above)
Comment (MichalBryxi) [reply docs/cli/add.md:87]: No hard feelings, I can revert it, but to show why I'm against copy&paste examples in inline form: Here you can see how precise I have to be to select _exactly_ what I intended:



Compare it to the code-fence version that has the _full line_ for mousing around and thus it's way harder to select unwanted things:

Comment (MichalBryxi) [reply docs/cli/add.md]: I pushed the revert to adhere to the ways it was. Let me know if it works for you now.

Comment (zkochan) [reply docs/cli/add.md]: But you could do it like this:

```
* Strict semver:
  
  `` `
  pnpm add zkochan/is-negative#semver:1.0.0
  `` `
```

Comment (MichalBryxi) [reply docs/cli/add.md:88]: Oooh, yeah why not. Pushed given change for both blocks. Also I used "Sentence case:" for the bullet points. Let me know if this works for you.

**Comment by MichalBryxi** [comment]
Preview looks reasonable (to me).

**Comment by zkochan** [comment]
merged https://github.com/pnpm/pnpm.io/commit/2be27c08dc0b44d9412e802a5cf40f55ee1ed494

**Pull request closed without merging**